cmd/snap/cmd_run.go

```diff
@@ -1132,6 +1132,16 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 		env["XAUTHORITY"] = xauthPath
 	}
 
+	// Guarantee that XDG_RUNTIME_DIR does exist before launching the snap
+	if xdg_runtime_dir, exists := env["XDG_RUNTIME_DIR"]; exists {
+		if _, err = os.Stat(xdg_runtime_dir); os.IsNotExist(err) {
+			if err = os.Mkdir(xdg_runtime_dir, 0700); err != nil {
```

Comment:
```suggestion
		if err = os.Mkdir(xdg_runtime_dir, 0700); err != nil && !errors.Is(err, os.ErrExist) {
```

(and drop one of the close braces below)

The stat() before mkdir() doesn't guarantee that you won't get an EEXIST error from mkdir() (consider two invocations of the snap, such that they run this code path simultaneously).